kasa/auth.py

```diff
@@ -0,0 +1,9 @@
+"""Authentication class for username / passwords."""
+
+
+class AuthCredentials:
```

Comment:
I think the default should be empty string rather than none.  Apparently kasa klap devices disconnected from the cloud have an empty string username and password.  Better to try to authenticate with this every time and report failures than have to handle the meaning of None values for username or password no?

Reply:
This fallback should be done in the klapprotocol class itself, having it default to `None` makes it clear that it's undefined rather than something the user inputted.